pkg/auth/providers/common/usermanager.go

```diff
@@ -48,3 +63,110 @@ func (u *userManager) SetPrincipalOnCurrentUser(apiContext *types.APIContext, pr
 func (u *userManager) GetUser(apiContext *types.APIContext) string {
 	return apiContext.Request.Header.Get(userAuthHeader)
 }
+
+func (h *userManager) EnsureUser(principalName, displayName string) (*v3.User, error) {
+	// First check the local cache
+	u, err := h.checkCache(principalName)
+	if err != nil {
+		return nil, err
+	}
+	if u != nil {
+		return u, nil
+	}
+
+	// Not in cache, query API by label
+	u, labelSet, err := h.checkLabels(principalName)
+	if err != nil {
+		return nil, err
+	}
+	if u != nil {
+		return u, nil
+	}
+
+	// Doesn't exist, create user
+	user := &v3.User{
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: "user-",
+			Labels:       labelSet,
+		},
+		DisplayName:  displayName,
+		PrincipalIDs: []string{principalName},
+	}
+
+	created, err := h.mgmt.Management.Users("").Create(user)
+	if err != nil {
```

Comment:
Should we check if error is `already exists` error? or there is no chance it could happen

Reply:
no chance it could happen. we are using a generated name, so creating a new user will always work